samcli/local/apigw/local_apigw_service.py

```diff
@@ -205,6 +201,58 @@ def _construct_error_handling(self):
         # Something went wrong
         self._app.register_error_handler(500, ServiceErrorResponses.lambda_failure_response)
 
+    def _generate_lambda_event(self, flask_request: Request, route: Route, method: str, endpoint: str) -> str:
+        """
+        Helper function to generate the correct Lambda event
+
+        Parameters
+        ----------
+        flask_request: Request
+            The global Flask Request object
+        route: Route
+            The Route that was called
+        method: str
+            The method of the request (eg. GET, POST) from the Flask request
+        endpoint: str
+            The endpoint of the request from the Flask request
+
+        Returns
+        -------
+        str
+            JSON string of event properties
+        """
+        # Default to None
+        # For Http Apis with payload version 1.0, API Gateway never sends the OperationName.
+        route_key = None
+
+        # TODO: Rewrite the logic below to use version 2.0 when an invalid value is provided
+        # the Lambda Event 2.0 is only used for the HTTP API gateway with defined payload format version equal 2.0
+        # or none, as the default value to be used is 2.0
+        # https://docs.aws.amazon.com/apigatewayv2/latest/api-reference/apis-apiid-integrations.html#apis-apiid-integrations-prop-createintegrationinput-payloadformatversion
+        if route.event_type == Route.HTTP and route.payload_format_version in [None, "2.0"]:
+            apigw_endpoint = PathConverter.convert_path_to_api_gateway(endpoint)
+            route_key = self._v2_route_key(method, apigw_endpoint, route.is_default_route)
+
+            return EventConstructor.v2_event_http(
+                flask_request=flask_request,
+                port=self.port,
+                binary_types=self.api.binary_media_types,
+                stage_name=self.api.stage_name,
+                stage_variables=self.api.stage_variables,
+                route_key=route_key,
+            )
+        elif route.event_type == Route.API:
```

Comment:
This can be changed to an `if` since we already return in the block above

Reply:
if there are multiple ifs and if the number of cases could increase in the future, a dispatch model would be great to adopt.

```python
for x in _:
    artifact = x.apply(artifact, *args, **kwargs)
```